### ISSUE TITLE
The option "allow404WithoutAuth" does not work if you have a route with a placeholder. Protected pages potentially leak.

### ISSUE THREAD

**Issue by ayalon** [sidebase/nuxt-auth#185]
### Environment

Nuxi 3.0.0                                                                                                   17:14:08
                                                                                                             17:14:09
RootDir: /app                                                                                                17:14:09
Nuxt project info:                                                                                           17:14:09

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.19.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@9.3.1`
- Builder:          `vite`

### Reproduction

Add a new route in the pages folder with a local auth middleware.
[language]
.. protected
 ....index.vue
```
<template>
  <div>
    <h1>Protected</h1>
  </div>
</template>

<script lang="ts" setup>
definePageMeta({ middleware: 'auth' })
</script>
```

**Nuxt.config.ts**
```
  auth: {
    enableGlobalAppMiddleware: false,
  },`
```

In the middleware we have this code:
```
  if (authConfig.globalMiddlewareOptions.allow404WithoutAuth) {
    const matchedRoute = router.getRoutes().find((route) => route.path === to.path);
```

`to.path` is in this example `/de/protected`
`route.path` is `/:language/protected`

The is no match and the auth middleware does not redirect and show the protected page.

### Describe the bug

If you use a local middleware and protect a route with a placeholder, the page will be displayed.
Adding  **allow404WithoutAuth: true** but there is a high chance that someone exposes protected pages with this check.
**Nuxt.config.ts**
```
  auth: {
    enableGlobalAppMiddleware: false,
    globalMiddlewareOptions: {
      allow404WithoutAuth: true,
    },
  },`
```

**Comment by BracketJohn** [comment]
>  Adding allow404WithoutAuth: true but there is a high chance that someone exposes protected pages with this check.

What page would leak in the above example?

**Comment by ayalon** [comment]
In the example above, you have a route resulting route for example '/de/protected'.
This page should be protected by middleware setting in the `definePageMeta`. But its freely accessible if the global middleware is disabled.

**Comment by ayalon** [comment]
@BracketJohn 
I have created a simple repro based on the nuxt-auth-example.
As long as the option `allow404WithoutAuth` is set to true, the URL` /de/protected` is shown, if false, you are redirected to the login page.

https://stackblitz.com/github/ayalon/nuxt-auth-example/tree/repro-protected-page?file=nuxt.config.ts

I how you can reproduce it with my example.

**Comment by BracketJohn** [comment]
Thanks for the repro! I implemented a fix that will be released soon, see #201 (:
